baseservice/iodevices/base/common.py

```diff
@@ -37,10 +42,21 @@ def bytes(self) -> bytes:
 
     @property
     def headers(self) -> MessageHeaders:
+        """
+        the headers for this message
+        """
         return self._headers
 
-    def copy(self):
-        return copy.copy(self)
+    def copy(self, new_headers: Optional[MessageHeaders] = None):
```

Comment:
Final is from 3.8 only...

Reply:
also, why won't it work?

Reply:
If i copy an inheritor of the class, i will not receive an instance of the inheritor.

As for typing.final, you may or may not choose to use typing_extentions, which contains backports of python's latest typing features. 

Will be understandable if you don't wish to add a new dependency, but just know that the option is available.

Reply:
Message is really basic, and i don't see why anyone would subclass it... although you're right. it will not work if someone will.

at this point, i don't want to use the Final extension (nor using the __init_subclass__ method for not allowing subclassing)